core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/stack/StackV4RequestValidator.java

```diff
@@ -89,17 +90,19 @@ private void validateSharedService(StackV4Request stackRequest, ValidationResult
         checkResourceRequirementsIfBlueprintIsDatalakeReady(stackRequest, validationBuilder);
         SharedServiceV4Request sharedService = stackRequest.getCluster().getSharedService();
         if (sharedService != null) {
-            Stack stack = stackService.getByNameInWorkspace(sharedService.getSharedClusterName(), restRequestThreadLocalService.getRequestedWorkspaceId());
-            if (stack == null) {
-                validationBuilder.error("Unable to attach to datalake because it doesn't exists.");
-            } else if (AVAILABLE.equals(stack.getStatus())) {
-                Optional<Cluster> cluster = Optional.ofNullable(clusterService.retrieveClusterByStackIdWithoutAuth(stack.getId()));
-                if (cluster.isPresent() && !AVAILABLE.equals(cluster.get().getStatus())) {
-                    validationBuilder.error("Ambari installation in progress or some of it's components has failed. "
-                            + "Please check Ambari before trying to attach cluster to datalake.");
+            try {
+                Stack stack = stackService.getByNameInWorkspace(sharedService.getSharedClusterName(), restRequestThreadLocalService.getRequestedWorkspaceId());
+                if (AVAILABLE.equals(stack.getStatus())) {
```

Comment:
when the stack is null then it will throw a `NullPointerException`

Reply:
stackService.getByNameInWorkspace(..) can't return null

On the other hand, we should return an Optional<Stack> (or any other type within on the other services) or annotate the method return value with @Nullable at least. (the first option is more straightforward to me)

In my opinion.